java/src/main/native/src/ColumnViewJni.cpp

```diff
@@ -288,6 +289,22 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_segmentedReduce(
   CATCH_STD(env, 0);
 }
 
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_segmentedGather(
+    JNIEnv *env, jclass, jlong source_column, jlong gather_map_list, jboolean nullify_out_bounds) {
+  JNI_NULL_CHECK(env, source_column, "source column view is null", 0);
+  JNI_NULL_CHECK(env, gather_map_list, "gather map is null", 0);
+  try {
+    cudf::jni::auto_set_device(env);
+    auto src_col = reinterpret_cast<cudf::lists_column_view *>(source_column);
+    auto gather_map = reinterpret_cast<cudf::lists_column_view *>(gather_map_list);
```

Comment:
Please cast these to a `cudf::column_view` and then create a `cudf::lists_column_view` from them. Otherwise there is no checking that the columns passed in are really lists. If they are not really bad things could happen.

Reply:
Done.